lib/routes/apex-legends/news.ts

```diff
@@ -0,0 +1,135 @@
+    });
+    if (!langEnum.has(lang)) {
+        throw new InvalidParameterError(`Invalid language: ${lang}`);
+    }
+    if (!typeEnum.has(type)) {
+        throw new InvalidParameterError(`Invalid type: ${type}`);
+    }
+    if (type !== 'latest') {
+        apiParams.append('typeSlug', type);
+    }
+    if (lang !== 'en') {
+        apiParams.append('locale', lang);
+    }
+    const apiUrl = `https://drop-api.ea.com/news-articles/pagination?${apiParams}`;
+    const response = await got(apiUrl);
+    const { items: newsItems, featured } = response.data;
+    type NewsItem = {
+        title: string;
+        description: string;
+        link: string;
+        pubDate: Date;
+        slug: string;
+    };
+    const allItems: NewsItem[] = [];
+    if (featured) {
+        allItems.push({
+            title: featured.title,
+            description: featured.summary,
+            link: `https://www.ea.com${lang === 'en' ? '/' : '/' + lang + '/'}games/apex-legends/apex-legends/news/${featured.slug}`,
+            pubDate: parseDate(featured.publishingDate),
+            slug: featured.slug,
+        });
+    }
+    allItems.push(
+        ...newsItems.map((item) => ({
+            title: item.title,
+            description: item.summary,
+            link: `https://www.ea.com${lang === 'en' ? '/' : '/' + lang + '/'}games/apex-legends/apex-legends/news/${item.slug}`,
+            pubDate: parseDate(item.publishingDate),
+            slug: item.slug,
+        }))
+    );
+    const items = await Promise.all(
+        allItems.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const response = await ofetch(`https://drop-api.ea.com/news-articles/${item.slug}${lang === 'en' ? '' : '?locale=' + lang}`);
+                item.description = md.render(response.body);
+                return item;
+            })
+        )
+    );
+    return {
+        title: `Apex Legends 官网资讯${type === 'latest' ? '' : `（${type === 'news-article' ? '最新消息' : '游戏更新'}）`}`,
+        link: `https://www.ea.com${lang === 'en' ? '/' : '/' + lang + '/'}games/apex-legends/apex-legends/news${type === 'latest' ? '' : `?type=${type}`}`,
+        item: items.map((item) => ({
+            title: item.title,
+            description: item.description,
+            link: item.link,
+            pubDate: item.pubDate,
+        })),
+        image: 'https://drop-assets.ea.com/images/F1GeiHWipvvKj7GtUVP3U/31bb122451e2dea6d14c9b497f8e09d4/apex-white-nav-logo.svg',
+        radar: [
+            {
+                source: ['www.ea.com/games/apex-legends/apex-legends/news'],
+                target: '/news',
+            },
+            {
+                source: ['www.ea.com/:lang/games/apex-legends/apex-legends/news'],
+                target: '/news/:lang',
+            },
+            {
+                source: ['www.ea.com/:lang/games/apex-legends/apex-legends/news?type=:type'],
+                target: '/news/:lang/:type',
+            },
```

Comment:
```suggestion
            },
```
source cant match search parameters